ambry-server/src/test/java/com.github.ambry.server/ServerTestUtil.java

```diff
@@ -1169,7 +1110,112 @@ protected static void endToEndReplicationWithMultiNodeSinglePartitionTest(String
+
+    int numTries = 4;
+    boolean foundTarget = false;
+    while (!foundTarget && numTries > 0) {
+      Thread.sleep(5000);
+      numTries--;
+      Set<String> setToCheck = new HashSet<String>(completeSetToCheck);
+      int numFound = 0;
+      for (String mountPath : mountPaths) {
+        File replicaTokenFile = new File(mountPath, "replicaTokens");
+        if (replicaTokenFile.exists()) {
+          CrcInputStream crcStream = new CrcInputStream(new FileInputStream(replicaTokenFile));
+          DataInputStream dataInputStream = new DataInputStream(crcStream);
+          try {
+            short version = dataInputStream.readShort();
+            Assert.assertEquals(version, 0);
+
+            System.out.println("setToCheck" + setToCheck.size());
+            while (dataInputStream.available() > 8) {
+              // read partition id
+              PartitionId partitionId = clusterMap.getPartitionIdFromStream(dataInputStream);
+              // read remote node host name
+              String hostname = Utils.readIntString(dataInputStream);
+              // read remote replica path
+              Utils.readIntString(dataInputStream);
+
+              // read remote port
+              int port = dataInputStream.readInt();
+              Assert.assertTrue(setToCheck.contains(partitionId.toString() + hostname + port));
+              setToCheck.remove(partitionId.toString() + hostname + port);
+              // read total bytes read from local store
+              dataInputStream.readLong();
+              // read replica token
+              FindToken token = factory.getFindToken(dataInputStream);
+              System.out.println(
+                  "partitionId " + partitionId + " hostname " + hostname + " port " + port + " token " + token);
+              ByteBuffer bytebufferToken = ByteBuffer.wrap(token.toBytes());
+              Assert.assertEquals(bytebufferToken.getShort(), 0);
+              int size = bytebufferToken.getInt();
+              bytebufferToken.position(bytebufferToken.position() + size);
+              long parsedToken = bytebufferToken.getLong();
+              System.out.println("The parsed token is " + parsedToken);
+              if (partitionId.isEqual(targetPartition)) {
+                Assert.assertFalse("Parsed offset must not be larger than target value: " + targetOffset,
+                    parsedToken > targetOffset);
+                if (parsedToken == targetOffset) {
+                  numFound++;
+                }
+              } else {
+                Assert.assertTrue("Offset should be 0 on unmodified partitions", parsedToken == -1);
+              }
+            }
+            long crc = crcStream.getValue();
+            Assert.assertEquals(crc, dataInputStream.readLong());
+          } catch (IOException e) {
+            Assert.fail();
+          } finally {
+            dataInputStream.close();
+          }
+        } else {
+          Assert.fail();
```

Comment:
Is this necessary? Is there a reason why this file *must* be present always. It should be eventually present though, but that gets checked as part of the `foundTarget` check, doesn't it?